apitools/base/py/testing/mock.py

```diff
@@ -345,15 +347,17 @@ def Unmock(self):
         del self._url
         del self._http
 
-        if self._request_responses:
-            raise ExpectedRequestsException(
-                [(rq_rs.key, rq_rs.request) for rq_rs
-                 in self._request_responses])
-        self._request_responses = []
-
         self.__client_class.IncludeFields = self.__real_include_fields
         self.__real_include_fields = None
 
+        requests = [(rq_rs.key, rq_rs.request)
+                    for rq_rs in self._request_responses]
+        self._request_responses = []
+
+        if requests:
+            if not supress and sys.exc_info()[1] is None:
```

Comment:
Why do you need the 'and sys.exc_info()[1] is None' check here? Isn't suppress already set appropriately using that check?